IPython/core/magics/basic.py

```diff
@@ -377,7 +379,24 @@ def xmode_switch_err(name):
             xmode_switch_err('user')
 
     @line_magic
-    def quickref(self,arg):
+    def pip(self, args=''):
+        """
+        Intercept usage of `pip` in the IPython shell and print an explanation.
+        """
+        print(textwrap.dedent('''
+        The following command is not meant to be run from a Python shell:
```

Comment:
I think this could be worded more clearly.

> pip commands should be run in a system shell or command prompt, not inside Python.

We could even check the platform and switch between 'system shell' for Unix and 'command prompt' for Windows.